constraint/pkg/client/e2e_test.go

```diff
@@ -834,24 +835,103 @@ func TestE2E_Tracing_Unmatched(t *testing.T) {
 	}
 }
 
-// TestE2E_Review_RegoEvaluationMeta tests that we can get stats out of evaluated constraints.
-func TestE2E_Review_RegoEvaluationMeta(t *testing.T) {
+func TestE2E_DriverCfg(t *testing.T) {
+	tests := []struct {
+		name         string
+		statsEnabled bool
+	}{
+		{
+			name:         "disabled",
+			statsEnabled: false,
+		},
+		{
+			name:         "enabled",
+			statsEnabled: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := context.Background()
+			c := clienttest.New(t)
+
+			_, err := c.AddTemplate(ctx, clienttest.TemplateDeny())
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			_, err = c.AddConstraint(ctx, cts.MakeConstraint(t, clienttest.KindDeny, "foo"))
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			obj := handlertest.Review{Object: handlertest.Object{Name: "bar"}}
+
+			rsps, err := c.Review(ctx, obj, drivers.Stats(tt.statsEnabled))
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			stats := rsps.StatsEntries
+			if stats == nil && tt.statsEnabled {
+				t.Fatal("got nil stats but stats enabled for Review")
+			} else if len(stats) != 0 && !tt.statsEnabled {
+				t.Fatal("got stats but stats disabled")
+			}
+
+			_, err = c.AddData(ctx, &obj.Object)
+			if err != nil {
+				t.Fatal(err)
+			}
+		})
+	}
+}
+
+// TestE2E_Review_StatsEntries tests that we can get stats out of evaluated constraints.
+// In particular, this test makes sure we have stats for both violating constraint kind, name pairs
+// and non violating ones.
+func TestE2E_Review_StatsEntries(t *testing.T) {
```

Comment:
This test should iterate over a list of test cases. Each test case should have an inbound object and an expected stats object.

This will make what is being tested and the expected result more clear.

Reply:
I'll be adding a tt style closer to the driver, in the `driver_unit_test.go`. I think this end to end is still valuable but has grown in scope and will trim it down to just show that stats are received in a follow up commit.